### ISSUE TITLE
GraphQL support

### ISSUE THREAD

**Issue by cyp3rius** [VirtusLab-Open-Source/strapi-plugin-comments#54]
1. Native support for GQL in Strapi
2. Research topic with Gatsby (https://www.gatsbyjs.com/plugins/gatsby-source-strapi-plugin-navigation/?=strapi)

References: #13, #24

**Comment by SalahAdDin** [comment]
Is there any ETA for this?

**Comment by cyp3rius** [comment]
@SalahAdDin first draft PR by @Kronos66 is already here.

**Comment by kwiat1990** [comment]
Hi @cyp3rius, any progress on this one? When we could expect this feature to be released?

**Comment by cyp3rius** [comment]
@kwiat1990  @SalahAdDin  -> `strapi-plugin-comments@1.0.2-beta.1` has been published recently and contains basic support for GQL

**Comment by pogosoftware** [comment]
Hello

I have strapi project with GraphQL enabled and after installing strapi-plugin-comments I receive error:

"GraphQLError: Syntax Error: Expected Name, found "}"."

I'm using strapi 3.6.6 and strapi-plugin-comments: ^1.0.2-beta.1

Any idea what I'm doing wrong there :( ?

This is a clean project without any custom modifications

**Comment by Kronos66** [comment]
Hi @pogosoftware 
I checked your error and the reason why you got this is not configured a comments plugin :(. When I create GQL schema I use config to add to schema relation content types:
```js
plugins.comments.relatedContentTypes
```
That configuration should contain an object with configuration content types in relation to the comments plugin. I forgot to add documentation to readme about this, but if you create this configuration everything should start working,. 
Example configuration:
```js
 comments: {
    relatedContentTypes: {
      pages: {
        contentManager: true,
        isSingle: true,
        key: 'title',
        value: 'id'
      },
      test: {
        contentManager: false
      }
    },
  },
```

**Comment by pogosoftware** [comment]
@Kronos66 thanks! It's working now.

But I found 2 other issues. 

First, if you block some comments then GraphQL return it. It would be nice if blocked comments and threads will be automatically filtered out from GraphQL query results.

Second, i tried to filter block comments using this query:
```
query GetArticles {
	articles(where: { comments: { blocked: false}}) {
		title
		comments {
			id
			content
			blocked
		}
	}
```
But i received and error:
```
"select distinct `articles`.* from `articles` left join `main`.`comments` as `comments_1` on `comments_1`.`related` = `articles`.`id` where `comments_1`.`blocked` = false and (`articles`.`published_at` is not null) limit 100 - SQLITE_ERROR: no such column: comments_1.related"
```

**Comment by Kronos66** [comment]
@pogosoftware thanks for the feedback!
I checked that problem and I think your query won't work correctly because you tried to filter on polymorphic relation where strapi have a problem with this relation type https://github.com/strapi/strapi/issues/4254
I prepared a special `query` for this can't use built-in methods

**Comment by sheddy123** [comment]
> 
> 
> Hi @pogosoftware
> I checked your error and the reason why you got this is not configured a comments plugin :(. When I create GQL schema I use config to add to schema relation content types:
> 
> ```js
> plugins.comments.relatedContentTypes
> ```
> 
> That configuration should contain an object with configuration content types in relation to the comments plugin. I forgot to add documentation to readme about this, but if you create this configuration everything should start working,.
> Example configuration:
> 
> ```js
>  comments: {
>     relatedContentTypes: {
>       pages: {
>         contentManager: true,
>         isSingle: true,
>         key: 'title',
>         value: 'id'
>       },
>       test: {
>         contentManager: false
>       }
>     },
>   },
> ```

Please where do I find this file.. location

**Comment by pogosoftware** [comment]
@sheddy123 I've created a plugin.js file inside the config directory with this content.
`
module.exports = ({ env }) => ({
  comments: {
      relatedContentTypes: {
        pages: {
          contentManager: true,
          isSingle: true,
          key: 'title',
          value: 'id'
        },
        test: {
          contentManager: false
        }
      },
    },
  });
`
That works for me. If you need more information abot that please go to the [official documentation](https://strapi.io/documentation/developer-docs/latest/development/plugins/email.html#configure-the-plugin)

**Comment by sheddy123** [comment]
> 
> 
> @sheddy123 I've created a plugin.js file inside the config directory with this content.
> 
> ```
> module.exports = ({ env }) => ({
>   comments: {
>       relatedContentTypes: {
>         pages: {
>           contentManager: true,
>           isSingle: true,
>           key: 'title',
>           value: 'id'
>         },
>         test: {
>           contentManager: false
>         }
>       },
>     },
>   });
> ```
> 
> That works for me. If you need more information abot that please go to the [official documentation](https://strapi.io/documentation/developer-docs/latest/development/plugins/email.html#configure-the-plugin)

I couldn't figure that as well. Trying to use REST Api rather. But calling the endpoint "http://localhost:1337/comments/article:1" I get this as my response
{
    "statusCode": 403,
    "error": "Forbidden",
    "message": "Forbidden"
}
Please help on how to resolve this.. and for path specification I would be glad if that's included

**Comment by cyp3rius** [comment]
@sheddy123 have you turned on the endpoints in settings? By default they are not exposed publicly.

**Comment by sheddy123** [comment]
> 
> 
> @sheddy123 have you turned on endpoints in the settings? By default they are not exposed publicly.

@cyp3rius I am new to this... so could you be more specific... The endpoint have been turned in the roles section(public) on the UI... so for the Comments part under permission, findall, findone, and put have been checked, but still get same error. I have an articles collection type and want to do a [GET and POST] http://localhost:1337/comments/article:1 for GET but returns 
{
    "statusCode": 403,
    "error": "Forbidden",
    "message": "Forbidden"
}
using POSTMAN... please I need more clarification on this thanks

**Comment by VuQuang** [comment]
> @sheddy123 I've created a plugin.js file inside the config directory with this content.
> 
> ```
> module.exports = ({ env }) => ({
>   comments: {
>       relatedContentTypes: {
>         pages: {
>           contentManager: true,
>           isSingle: true,
>           key: 'title',
>           value: 'id'
>         },
>         test: {
>           contentManager: false
>         }
>       },
>     },
>   });
> ```
> 
> That works for me. If you need more information abot that please go to the [official documentation](https://strapi.io/documentation/developer-docs/latest/development/plugins/email.html#configure-the-plugin)

I use name **plugins.js** instead of **plugin.js** and worked.

**Comment by SalahAdDin** [comment]
> > @sheddy123 I've created a plugin.js file inside the config directory with this content.
> > ```
> > module.exports = ({ env }) => ({
> >   comments: {
> >       relatedContentTypes: {
> >         pages: {
> >           contentManager: true,
> >           isSingle: true,
> >           key: 'title',
> >           value: 'id'
> >         },
> >         test: {
> >           contentManager: false
> >         }
> >       },
> >     },
> >   });
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > That works for me. If you need more information abot that please go to the [official documentation](https://strapi.io/documentation/developer-docs/latest/development/plugins/email.html#configure-the-plugin)
> 
> I use name **plugins.js** instead of **plugin.js** and worked.

actually, the correct name is `plugins.js`.

**Comment by cyp3rius** [comment]
Feature has been released as `v1.0.2`

**Comment by SalahAdDin** [comment]
@cyp3rius So, it means it is full operative now, right?
